open-craft/cohort-manager
# SE-1406 support cohorts pagination

The cohorts list API actually uses pagination. Unfortunately the pagination metadata is [stripped out](https://github.com/open-craft/edx-platform/blob/c3b9bb727c77698f0900bc52b576f6c87e730c15/openedx/core/djangoapps/course_groups/views.py#L513) in the response, so we didn't discover this originally.

This adds support for retrieving all pages of cohorts. It's not pretty, because the pagination metadata isn't exposed in the api response.

It also updates js dependencies (security vulnerabilities and why not) and the readme (was invalid).

**Test instructions**:

- install in lms following readme instructions
- run `make bundle` and verify that the bundle doesn't change (not sure how reproducible the builds are with rollup though)
-  temporarily set the page_size parameter in getCohorts to a small number (eg. 2)
- `make bundle`
- set up a course with cohorts - at least twice as many cohorts as the page_size parameter above
- open the cohort manager url (eg. http://localhost:18000/cohort-manager/?course_id=course-v1:edX%2BDemoX%2BDemo_Course ) or navigate to it from the instructor dashboard.
- verify that all cohorts load for the test course in the left pane
- paste the test course ID in the right text input and click 'load cohorts'
- verify that all cohorts load in the right pane
